### PR TITLE
Initial commit to mci-https-e2e recipe and changes in cluster-setup.md to support ASM

### PR DESCRIPTION
**Pull request by hashmisf**
Initial commit to mci-https-e2e recipe and changes in cluster-setup.md to provision a cluster that satisfies ASM minimum requirements

### CODE REVIEW

File: cluster-setup.md [hunk 1]
```diff
@@ -21,7 +27,9 @@ The single-cluster examples use the following GKE setup for deploying the manife
 
 The multi-cluster examples use the following GKE setup for deploying the manifests. If you've already created `gke-1` in the [single Cluster Section](#single-cluster-environment), you can reuse that cluster.
 
-1. Deploy two GKE clusters within your Google Cloud project.
+1. Deploy two GKE clusters within your Google Cloud project.  
+Note: Anthos Servish Mash has [some minumum requirements](https://cloud.google.com/service-mesh/v1.7/docs/scripted-install/gke-asm-onboard-1-7#requirements) for GKE.
+  Execute below command to create cluster if you don't need to install Anthos Service Mesh in your cluster later.
```
Comment (boredabdel) [inline cluster-setup.md:32]: I think we should provide a single way to create a multi cluster environment regardless of if ASM is needed or not. You can go ahead and remove the instructions for non ASM cases and keep the one for ASM

**Review by boredabdel — COMMENTED** [review]
One comment. the Rest looks good

File: cluster-setup.md (hunk 1 above)
Comment (hashmisf) [reply cluster-setup.md]: I have made updates as requested, please review now.

**Review by boredabdel — APPROVED** [review]
LGTM

**Pull request merged**